### ISSUE TITLE
Time-consuming uploads break

### ISSUE THREAD

**Issue by manuth** [pydio/cells#132]
Hello, Pydio-Team
I've been uploading a very large file - about 30 GBs.
Pydio automatically splits them into smaller chunks and uploads them one by one.

First I found myself unable to upload the file because the session kept closing while idling on my pydio-website.

For that reason I tried to keep browsing through folders in order to keep the session alive.
When doing so the upload keeps starting fron scratch again after some time.

**Comment by cdujeu** [comment]
Hi @manuth thanks for reporting
About session expiration, we had something for that P8, I guess it still has to be reported in the new code.
About the parts starting from scratch, maybe you can try to tweak the upload timeout in the Settings. This is a client-side timeout (JS) that may need to be longer.
-c

**Comment by cdujeu** [comment]
Should be fixed in 1.6

**Comment by jbaranek** [comment]
This is still a problem in 1.6.1.  Tried increasing frontend/plugin/gui.ajax/CLIENT_TIMEOUT to 120m, core.uploader/MULTIPART_UPLOAD_TIMEOUT_MINUTES to 120m, services/pydio.grpc.auth/dex/expiry/idTokens to 120m, staticClients/IdTokensExpiry to 120m to no avail.

Symptoms are as described in the original post: in slightly more detail - uploading a large file (I tested with 50GB) the upload restarts after ~10min of upload.  In the console, the upload request (https://....../io/personal-files/filename.dat?partNumber=492&uploadID=xxxx) fails with a 403.
On a side note, the failed uploads leave orphaned files in .minio.sys/multipart and .minio.sys/tmp that seem never to be purged.

I believe therefore this issue should be reopened.

Janusz

**Comment by cdujeu** [comment]
@jbaranek 
I wonder, did you actually try to raise the Multipart Upload Parts size ? Maybe for such big files, it would be more efficient to create less parts ...
Also did you restart and properly reload the browser to make sure the new settings were taken into account?
-c

**Comment by jbaranek** [comment]
Hi Charles,
Thanks for the reply.  Although on previous tests I restarted the browser, and cleared caches, I omitted to do so on the last tests when I increased the token timeouts - only restarting the server and exiting the session.
I have just repeated the tests.  With services/pydio.grpc.auth/dex/expiry/idTokens set to 120m and staticClients/IdTokensExpiry to 120m, the upload completes.  Resetting these to the default timeouts of 10min causes uploads to fail with a 403 in the browser log after 10 minutes.  So the workaround (increasing token expiry time) works, but shouldn't be necessary.

I was under the impression that tokens are automatically renewed (commits 2eaece8 and ffa7040, v1.2.4), but this is perhaps not the case, at least in my use case (uploading via a drag/drop to a logged in workspace - not a minisite)
Janusz

**Comment by cdujeu** [comment]
Hi, 
The should indeed does refresh itself (if required) before starting each part upload. Maybe this part does not work. At the end how long does your working upload take?

**Comment by jbaranek** [comment]
Around 20min for the working upload.
j

**Comment by jbaranek** [comment]
@cdujeu In which release was this fixed? (without the large lifespan workaround)

**Comment by bures** [comment]
It seems we are experiencing a similar problem. When a big file gets uploaded, it times out after several chunks complete well. It seems like some token does not get refreshed or similar. How should I increase the timeout values mentioned above? Or anything that could help. If you could please point me in the right direction, I could possibly even debug the code myself if needed. Thanks.

I have this Pydio Cells:
Pydio Cells Home Edition
 Version: 	4.2.5
 Built: 	17 Jul 23 09:04 +0000
 Git commit: 	0b98e8a5e7976e77964fab784b123ac55971aa04
 OS/Arch: 	linux/amd64
 Go version: 	go1.19.11

**Comment by bsinou** [comment]
hi @bures 
> If you could please point me in the right direction, I could possibly even debug the code myself if needed. 

Thanks for the offer. The right direction here is to rather start with describing your issue first in the [forum](https://forum.pydio.com) and avoid to resurrect old closed issues :) 
In addition to the info you already gave, you might also add details about the size of your file, the upload bandwidth you have and so on.

We will definitively open a new issue if necessary once we have dug further, since 2020 when we closed the issue, the code has changed **a lot**.

Regards

**Comment by bures** [comment]
Thanks. I'll check it thoroughly first and I will ask on the forum if necessary.
